Hipo/cra-template-hipo-typescript
# feat(template): create initial template

## Description
Forked from official typescript template, added some of our common components and configurations.
Includes:
- Basic folder structure
- Fundamental CSS overrides
- Router and two routes as an example
- Page components
- Linter configurations and dependencies
- Pre-commit hook with Husky

## Notes
- To examine the code and folder structure easily, I strongly suggest using GitHub editor: https://github.dev/Hipo/cra-template-hipo-typescript/tree/feat/init-template

## How to test on local?
You can clone the repo, checkout to this branch, then go up one direction, and run:
```sh
npx create-react-app cra-test-app --template file:./cra-template-hipo-typescript
```
This should work just like default cra template.

## To be discussed
These are not included atm, I suggest we add them (in a next PR), we can discuss:
- Async process/component
    - Maybe we can add `useAsyncProcess` and use GitHub API (it's free) to fetch and show Hipo Org. information
      - api url: [https://api.github.com/orgs/hipo](https://api.github.com/orgs/hipo)
- App context
    - Maybe we can implement app state with a simple login logic. User can enter name and login by clicking on a button.

You can write if you have any additional suggestions
